lib/hci-socket/gatt.js

```diff
@@ -218,6 +218,15 @@ class Gatt  extends EventEmitter {
 
         for (let k = 0; k < characteristic.descriptors.length; k++) {
           const descriptor = characteristic.descriptors[k];
+          
+          //https://github.com/abandonware/bleno/issues/51
+          //above, if the characteristic defines notify, the 2902 characteristic descriptor is added automatically
+          //if the peripheral also declares this, we don't want to add it again
+          //flags (properties, secure etc) above are better defined
+          if(descriptor.uuid == '2902')
```

Comment:
is there a constant name to be defined for 2902 ?

Reply:
There was an existing constant `GATT_CLIENT_CHARAC_CFG_UUID` defined further up at line 41, but this is declared as hex 0x2902 and we need it as a string for both the new change and the descriptor being added slightly further up.

I have changed this to declare a new constant `GATT_CLIENT_CHARAC_CFG_UUID_S` and used it in both places. Is that what you meant?